Service module fails to execute service command

Hi,
Having issue with running service command use from testinfra - it fails to be executed properly.

Example invocation:
`ctx.Service('ntpd').is_running`

I am getting this command executed actually:
`$ ssh -o User=<user> -- <server>.com 'service ntpd status'`

and this output from a particular server:
`bash: service: command not found`

I see from code from where this is being executed:
```
class SysvService(Service):

    @property
    def is_running(self):
        # based on /lib/lsb/init-functions
        # 0: program running
        # 1: program is dead and pid file exists
        # 3: not running and pid file does not exists
        # 4: Unable to determine status
        return self.run_expect(
            [0, 1, 3], "service %s status", self.name).rc == 0
```

My further investigation shows that when I do login, and bash profile is loaded - then I can execute service command successfully from the server in question.
Running from Testinfra works differently - it triggers a command invocation from single ssh request, so I can see that no path to service is found and command fails. 
I also see that running command with this format will be successfull:
`$ ssh -o User=<user> -- <server>.com '/sbin/service ntpd status'`

Here is actual traceback, (with removed sensitive data):

`Traceback (most recent call last):
  File "<...>\tests\example\sh_test.py", line 17, in test
    assert ctx.Service('ntpd').is_running
  File "<...>\testinfra\modules\service.py", line 85, in is_running
    [0, 1, 3], "service %s status", self.name).rc == 0
  File "<...>\testinfra\modules\base.py", line 34, in run_expect
    pytest.fail("Unexpected exit code %s for %s" % (out.rc, out))
  File "c:\python27\lib\site-packages\_pytest\runner.py", line 542, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Unexpected exit code 127 for CommandResult(command="ssh -o User=<...> -- <...>.com 'service ntpd status'", exit_status=127, stdout=u'', stderr='bash: service: command not found\n')`

Note that we are using modifed testinfra but I believe root cause is still the same as in original sources.

Hello, I am runnign an issue simalary of yours

```
    =================================== FAILURES ===================================
    ______________ test_ntpd_enabled[ansible://centos6] ______________
    
    Service = <class 'testinfra.modules.base.SysvService'>
    
        def test_ntpd_enabled(Service):
            s = Service('ntpd')
    >       assert s.is_running
    
    tests/test_default.py:12:
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    
    self = <service ntpd>
    
        @property
        def is_running(self):
            # based on /lib/lsb/init-functions
            # 0: program running
            # 1: program is dead and pid file exists
            # 3: not running and pid file does not exists
            # 4: Unable to determine status
            return self.run_expect(
    >           [0, 1, 3], "service %s status", self.name).rc == 0
    E       AssertionError: Unexpected exit code 127 for CommandResult(command=u'service ntpd status', exit_status=127, stdout=u'', stderr=u'/bin/sh: service: command not found')
    
    ../../venv/lib/python2.7/site-packages/testinfra/modules/service.py:80: AssertionError

```

I'm using the centos/6 vagrant image
It's a path issue
I have modified service.py in order to pass the test : 
```
-              [0, 1, 3], "service %s status", self.name).rc == 0
+              [0, 1, 3], "PATH=\"$PATH:/bin:/sbin\" service %s status", self.name).rc == 0
```

--- a/testinfra/modules/service.py
+++ b/testinfra/modules/service.py
@@ -69,6 +69,24 @@ class Service(Module):
 
 class SysvService(Service):
 
+    def __init__(self, name):
+        super(SysvService, self).__init__(name)
+        self._command = None
+
+    @property
+    def _service_command(self):
+        if self._command is None:
+            if self._host.exists('service'):
+                self._command = 'service'
+            # service command may not exist in PATH for non privileged users
+            elif self._host.file('/sbin/service').exists:
+                self._command = '/sbin/service'
+            elif self._host.file('/usr/sbin/service').exists:
+                self._command = '/usr/sbin/service'
+            else:
+                raise RuntimeError('cannot find "service" command')
+        return self._command
+
     @property
     def is_running(self):
         # based on /lib/lsb/init-functions
@@ -77,7 +95,8 @@ class SysvService(Service):
         # 3: not running and pid file does not exists
         # 4: Unable to determine status
         return self.run_expect(
-            [0, 1, 3], "service %s status", self.name).rc == 0
+            [0, 1, 3], "%s %s status",
+            self._service_command, self.name).rc == 0
 
     @property
     def is_enabled(self):

